### PR TITLE
chore: Update accounts-controller @v19.0.0 and keyring-controller @v18.0.0

### LINKED ISSUE #12302: Update metamask/accounts-controller to version 19.0.0
Please update metamask/accounts-controller to version 19.0.0

### LINKED ISSUE #12304: Update metamask/keyring-controller to version 18.0.0
Please update metamask/keyring-controller to version 18.0.0

### PR DESCRIPTION
**Pull request by owencraston**
## **Description**
This PR updates the accounts controller to the latest version of 19. This updates requires a peer dependancy bump of the keyring controller to version 18.0.0

- See accounts controller [changelog](https://github.com/MetaMask/core/blob/main/packages/accounts-controller/CHANGELOG.md#1900).
- See keyring controller [changelog](https://github.com/MetaMask/core/blob/main/packages/keyring-controller/CHANGELOG.md#1800).
    - version 18 of the keyring controller removes the [addNewAccountWithoutUpdate](https://github.com/MetaMask/core/pull/4845) which is used once in the mobile repo. In order to merge this PR we will need to wait for @mikesposito's [PR to merge](https://github.com/MetaMask/metamask-mobile/pull/11409). This PR removes the need for `addNewAccountWithoutUpdate` and improves performance/reliability. 

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-mobile/issues/12302
Fixes: https://github.com/MetaMask/metamask-mobile/issues/12304

Unblocked by: https://github.com/MetaMask/metamask-mobile/pull/11409

#### Account creation

1. open app
2. press create wallet 
3. go through the onboarding flow
4. once onboarding is complete, you should be on the wallet view and see a single account
5. open the account selector list and add a new account'
6. this should work without issues
7. change the name of your new account
8. force close the app and reopen, the previously selected account should be the same and the custom name should be preserved.

#### Account importing

1. assuming you have created a wallet wth the above steps....
2. open the account selector list and click add account or hardware wallet and then click import account
3. paste a private key in this text field and then import
4. the account should be imported, selected and have the label "imported"
5. force closing the app and re opening it should preserve the same state of accounts

#### Hardware wallets

1. assuming you have created a wallet wth the above steps....
2. open the account selector list and click add account or hardware wallet and then click Add hardware wallet
3. select your hardware wallet of choice (ledger or qr based)
4. follow the instructions to add a hardware wallet
5. the steps should work and you should have successfully added a hardware account
6. the hardware account should have the correct labels (either Qr hardware or ledger)
7. force closing the app and reopening should preserve the same accounts state. 

### **After**

https://github.com/user-attachments/assets/ab8ca48e-dc48-4601-a56c-5c6d7fabab3e

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -4724,16 +4735,16 @@
...
   dependencies:
-    "@metamask/base-controller" "^6.0.2"
-    "@metamask/controller-utils" "^11.0.2"
-    "@metamask/json-rpc-engine" "^9.0.2"
-    "@metamask/rpc-errors" "^6.3.1"
-    "@metamask/utils" "^9.1.0"
+    "@metamask/base-controller" "^7.0.2"
+    "@metamask/controller-utils" "^11.4.1"
+    "@metamask/json-rpc-engine" "^10.0.1"
+    "@metamask/rpc-errors" "^7.0.1"
```
Comment (Cal-L) [inline yarn.lock:4746]: Let's ensure that this bump doesn't crash the app. We have patches for rpc errors that fixed issues in the past

**Review by Cal-L — COMMENTED** [review]
Could you also sync with @tommasini on these changes since he's working on a performance optimization, which includes a keyringcontroller patch

File: app/core/Engine/Engine.ts [hunk 2]
```diff
@@ -62,7 +62,7 @@ import {
 } from '@metamask/snaps-controllers';
 
 import { WebViewExecutionService } from '@metamask/snaps-controllers/react-native';
-import { NotificationParameters } from '@metamask/snaps-rpc-methods/dist/restricted/notify.cjs';
+import { NotificationArgs } from '@metamask/snaps-rpc-methods/dist/restricted/notify.cjs';
```
Comment (tommasini) [inline app/core/Engine/Engine.ts:65]: Just to check, this change was not needed to update accounts and keyring controllers right? Just noticing this seems to be a kind side touch

**Review by tommasini — APPROVED** [review]
LGTM!

Yeah, the Keyring Controller patch will need to be applied to this version as well!

File: app/core/Engine/Engine.ts (hunk 2 above)
Comment (owencraston) [reply app/core/Engine/Engine.ts:65]: This type no longer exists in the newer version of this package but the signature of this type is the same. I verified that this is what the snaps package is looking for.

File: yarn.lock (hunk 1 above)
Comment (Gudahtt) [reply yarn.lock:4746]: Good callout, but there is a resolution in place for this package that ensures we stay on one synchronized version, so we're good here

Comment (owencraston) [reply yarn.lock:4746]: I can confirm that I tested the app and did not experience any crashes. Looking at [this PR](https://github.com/MetaMask/metamask-mobile/pull/12428) we will need to update the patches once this merges.

**Comment by owencraston** [comment]
Simulator Screenshot - iPhone 13 mini - 2024-11-27 at 15 36 04


confirmed that the dapp interactions seem to be working as expected.

**Pull request merged**